tests/post/steps/test_liveness.py

```diff
@@ -27,84 +30,54 @@ def test_expected_pods(host):
     "empty in the '{namespace}' namespace"))
 def check_resource_list(host, resource, namespace):
     with host.sudo():
-        cmd = ("kubectl --kubeconfig=/etc/kubernetes/admin.conf"
-               " get {0} --namespace {1} -o custom-columns=:metadata.name")
-        cmd_res = host.check_output(cmd.format(resource, namespace))
-    assert len(cmd_res.strip()) > 0, 'No {0} found in namespace {1}'.format(
+        output = host.check_output(
+            "kubectl --kubeconfig=/etc/kubernetes/admin.conf "
+            "get %s --namespace %s -o custom-columns=:metadata.name",
+            resource,
+            namespace,
+        )
+
+    assert len(output.strip()) > 0, 'No {0} found in namespace {1}'.format(
             resource, namespace)
 
 
 @then(parsers.parse(
     "we can exec '{command}' in the pod labeled '{label}' "
     "in the '{namespace}' namespace"))
 def check_exec(host, command, label, namespace):
-    candidates = _get_pods(host, label, namespace)
-    try:
-        pod, = candidates
-    except ValueError:
+    candidates = kube_utils.get_pods(host, label, namespace)
+
+    if len(candidates) != 1:
```

Comment:
Instead of `if`, `pytest.fail`, `return`, why not

```python
assert len(candidates) == 1, "Expected one (..."
```

Reply:
Indeed, I just converted the `try/except` to this but an `assert` is more idiomatic for `pytest`